src/Build/Logging/FileLogger.cs

```diff
@@ -39,6 +39,11 @@ public FileLogger()
                 colorReset: BaseConsoleLogger.DontResetColor)
         {
             WriteHandler = Write;
+
+            if (EncodingUtilities.GetExternalOverridenUILanguageIfSupportableWithEncoding() != null)
```

Comment:
I think there's a case where this won't work quite right:

1. The user starts a build with `DOTNET_CLI_UI_LANGUAGE=ja`
2. The build completes
3. the user unsets the environment variable, but does not kill worker nodes
4. The user starts a new build with `-distributedFileLogger`, which will run this `FileLogger` ctor but with a stale environment.

Ideally we'd peek into `NodeConfiguration.BuildParameters.Culture` and see if it's representable in the default codepage, but it doesn't seem very easy to plumb that down to this layer.

Reply:
This is a good thing I did not consider, thanks. I took this offline but when I tested it, it seemed to work. Let's discuss what we should do

Reply:
It seems that it is a subtle bug in that the log will use utf 8 if you do the above steps with /m and also have to start on a non default system os language per @rainersigwald. We don't anticipate this to be bad as the logs will still be readable as utf-8 supports all encodings. In fact I think this is better, as if the fancylogger or other loggers add emojis, then they will show up correctly, with some native windows chcp they would not.

Reply:
spiderman_pointing_at_spiderman.gif :)